test/test-debuglet.ts

```diff
@@ -87,6 +87,28 @@ describe('CachedPromise', function() {
       done();
     });
   });
+
+  it('CachedPromise should return the same promise if not staled', (done) => {
```

Comment:
I don't think it's part of the public contract of `CachedPromise` to return same/difference `Promise` objects based on staleness. What I would expect of `CachedPromise` instead, is that if the underlying resource is deemed "fresh", then the `Promise` object we get from `CachedPromise#get` is already resolved; otherwise, it will not be resolved until `CachedPromise#resolve` is called. I think this is the behavior that should be tested... what do you think?

Reply:
Done.